rust/lance-encoding/src/compression.rs

```diff
@@ -132,13 +135,36 @@ impl CompressionStrategy for DefaultCompressionStrategy {
                 // size might be smaller than the compressed size.
                 let too_small = bit_widths.len() == 1
                     && InlineBitpacking::min_size_bytes(bit_widths.value(0)) >= data.data_size();
-                if !has_all_zeros
-                    && !too_small
-                    && (fixed_width_data.bits_per_value == 8
-                        || fixed_width_data.bits_per_value == 16
-                        || fixed_width_data.bits_per_value == 32
-                        || fixed_width_data.bits_per_value == 64)
+
+                if let Some(compression) = field.metadata.get(COMPRESSION_META_KEY) {
```

Comment:
cc @westonpace, I introduce this so that user can enforce an encoding for `rle` verus `bitpacking`. Let me know what do you think?

Reply:
This idea looks great to me.

Reply:
I created an issue https://github.com/lancedb/lance/issues/4204 to track this.